### PR TITLE
Build packages as both CJS and ESM

### PR DESCRIPTION
**Pull request by Mrtenz**
- We now use SWC to transpile TypeScript to JavaScript (both ESM and CJS).
- TypeScript (`tsc`) is still used to generate types (and thus perform type checking).
- The build job in CI was split into two jobs, to perform the two above tasks separately.
- Each `package.json` now defines an `exports` field, so that consumers of the libraries can choose the right version for their specific environment.

### CODE REVIEW

**Comment by Mrtenz** [comment]
@socket-security ignore-all

**Comment by Mrtenz** [comment]
@metamaskbot publish-preview

**Comment by Mrtenz** [comment]
@metamaskbot publish-preview

**Comment by Mrtenz** [comment]
@metamaskbot publish-preview

**Comment by Mrtenz** [comment]
I have verified that this works in the MetaMask extension.

File: packages/create-snap/package.json [hunk 1]
```diff
@@ -8,15 +8,30 @@
...
+    ".": {
+      "import": "./dist/esm/index.js",
+      "require": "./dist/cjs/index.js",
+      "types": "./dist/types/index.d.ts"
+    }
+  },
+  "main": "dist/cjs/index.js",
+  "module": "dist/esm/index.js",
+  "types": "dist/types/index.d.ts",
+  "bin": "dist/cjs/main.js",
```
Comment (FrederikBolding) [inline packages/create-snap/package.json:21]: Does this need to be `./` prefixed? 🤔

Comment (Mrtenz) [reply packages/create-snap/package.json:21]: I don't think it's necessary, but I did it anyway for consistency.

**Comment by Mrtenz** [comment]
@SocketSecurity ignore-all

**Pull request merged**